### ISSUE TITLE
[EN][Spell] AMS vs Seduction

### ISSUE THREAD

**Issue by Flinvoltx** [Atlantiss/BugTracker#10200]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

[//]: # (On the Bugtracker all titles of reports must be written in English so that everyone, no matter which language they use everyday, could search and check if the problem already has been reported by another player. When describing the problem you can use English as well as Polish and in nearest future - others. It is important to determine at the beginning of a title in which language it is written by using symbols [EN] or [PL].)

[//]: # (Enclose links to things related to the bug using http://wowhead.com or http://cata.openwow.com)
[//]: # (Write your tickets according to the format:)
[//]: # ([EN][Quest][Zone] A Vision of the Past)
[//]: # ([EN][NPC] Thoralius the Wise)
[//]: # ([EN][Spell][Class] Frostfire Bolt)
[//]: # ([PL][Talent][Class] Brain Freeze)
[//]: # ([EN][Glyph][Class] Glyph of Frostfire Bolt)
[//]: # ([EN][Npc][Drop] Thoralius the Wise)
[//]: # ([EN][Web] Armory doesnt work)

**Description**: Channeling of Seduction (warlock pet spell) casted on Anti-Magic shield (DK spell) doesn't stops after stated that DK is immune to effect and continues casting until pet is stopped by recalling it.

**Current behaviour**: Channeling of Seduction (warlock pet spell) casted on Anti-Magic shield (DK spell) doesn't stops after stated that DK is immune to effect and continues casting until pet is stopped by recalling it.
Pet continues to cast it and nothing but a visual effect happens to DK. This makes controlling pet more difficult as it bugs out and won't cast another seduct on same target, because it is already casts it.

**Expected behaviour**: Channeling should be interrupted properly as it happens when casted on rogue with CoS.

**Steps to reproduce the problem**:

1. Order to cast seduct on DK
2. Enable AMS
3. Watch some movie

**Revision**: 5860

**Comment by Axelitoo** [comment]
Let's get something straight: Cloak of Shadows and Anti-Magic Shell do **_NOT_** behave the same way. You cannot remove magic effects with it, you just *prevent* them. 
So if I understand your report correctly, you should not be able to remove existing magic effects using AMS, however if you use AMS before they occur, then they should not work.
However, if it doesn't matter if you have AMS up or not, then you should find a video from retail proving that what you're saying is true.

**Comment by Flinvoltx** [comment]
Prevent and cancel channeling, seduct shouldn't be keep channeled when **prevented**. 
This is obvious bug and doesn't need proof IMO. Proof needs only what happens after AMS if channeling is not prevented.
![default](https://user-images.githubusercontent.com/30041703/44986084-36d45480-af8b-11e8-86f2-c7f585d39eea.png)
This shouldn't happen (Dk moves after AMS, channeling continues)

**Comment by Flinvoltx** [comment]
It doesn't prevent Drain Soul from keep channeled as well. So i guess the issue may be rename as "AMS vs Channeled abilities"

**Comment by Psychobird7** [comment]
"This is obvious bug and doesn't need proof IMO."           

That is simply not a thing, you need to bring proof of its proper behaviour if you want a fix.      

There is also no reason AMS should stop Drain soul from channeling.

**Comment by Axelitoo** [comment]
Okay so, I will quote tooltips this time to make it a *bit* more understandable.

Cloak of Shadows: "**_Instantly removes all existing harmful spell effects_**..". This indicates that it removes ANY sort of magical debuffs, including channeled spells.

Anti-Magic Shell: "**_...preventing application of harmful magical effects._** This however, tells you that it PREVENTS the application, it does not REMOVE any existing ones. As you know, channeled spells apply a debuff on the target, which makes it so that it won't interrupt the channel since the debuff application happend.

TL;DR: CoS removes debuffs, AMS prevents them. Big difference.

**Comment by Flinvoltx** [comment]
>...which makes it so that it won't interrupt the channel since the debuff application happend.
Thing is that debuff doesn't apply at all.
Looking for some proofs is tiresome. Found only this, which indicates that AMS is broken and needs to be reviewed carefully.
https://youtu.be/pj1I2d9gvVs?t=2m56s
against this on Atlantiss
![default](https://user-images.githubusercontent.com/30041703/44992513-11057a80-afa0-11e8-8e61-89a92f4f97b2.png)

**Comment by Flinvoltx** [comment]
>There is also no reason AMS should stop Drain soul from channeling.

Yes, but it should deal reduced damage and deal full damage after AMS expires.

**Comment by Flinvoltx** [comment]
https://youtu.be/-qemGMaPNcg
How it works on Atlantiss vs How it works on another server (WotLK, but i guess there should no big difference. At least Mind Flay bug already provided and has same problem).

**Comment by Flinvoltx** [comment]
>/vote for close.
any counterproofs for mine? Or just /close and leave DK positively broken?

**Comment by Flinvoltx** [comment]
I did already, a video from bg is from retail, upload date 2011 and there was no good non-retail cata servers that time, so it is 99.95% from retail. Clear evidence of existence of debuff application there and dealing damage. Mentioned topic about AMS not absorbind damage at all, while mine is about AMS blocks all damage while it shouldn't.

**Comment by xolon** [comment]
REEP LOX AND DKS
